### PR TITLE
Migration: Migrate org switcher to react

### PR DESCRIPTION
**Pull request by kaydelaney**
**What this PR does / why we need it**:
Migrate org switcher to react as part of the ongoing angular -> react migration.

### CODE REVIEW

**Comment by kaydelaney** [comment]
Leaving this comment as a reminder to myself to implement scrolling for the modal content before this gets merged

File: packages/grafana-data/src/types/orgs.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export interface UserOrgDTO {
+  orgId: number;
+  name: string;
+  role: Role;
+}
+
+export enum Role {
```
Comment (torkelo) [inline packages/grafana-data/src/types/orgs.ts]: This is already defined in https://github.com/grafana/grafana/blob/f58ec04c12d4678aaea04a9ab0325358e4d212b2/public/app/types/acl.ts#L1

Comment (kaydelaney) [reply packages/grafana-data/src/types/orgs.ts]: Thanks for pointing this out!

Comment (kaydelaney) [reply packages/grafana-data/src/types/orgs.ts]: Do you have any thoughts on whether the types defined in `grafana/public/app/types/acl.ts` should be moved to `@grafana/data`?

Comment (torkelo) [reply packages/grafana-data/src/types/orgs.ts]: some like OrgRole should I think, not sure about the rest

**Comment by torkelo** [comment]
Could this picked up an completed? Trying to reduce number of open PRs :)

**Comment by torkelo** [comment]
Would love to see this rebased, looks like it was very close to mergeable?

File: packages/grafana-data/src/types/orgs.ts (hunk 1 above)
Comment (torkelo) [reply packages/grafana-data/src/types/orgs.ts]: @ryantxu  what's your thoughts on http api types, where should they be defined? app, grafana-data, or grafana-runtime . 

@kaydelaney can you rename it to OrgRole so it matches the one define in acl ?

**Review by torkelo — APPROVED** [review]
Made some big changes to the Modal, felt there was pretty major duplication in how we send in the title using markup (and mixing new and old styles). 

The Modal also used cx function everywhere this could be removed, cx function is only needed when you 
need to mix old regular classNames and emotion styles. 

Review changes here: 
https://github.com/grafana/grafana/pull/19607/commits/3518c99f1337160f158bf5d9bfceec3883c5e93f

**Pull request merged**